app/src/main/java/com/cornellappdev/coffee_chats_android/ClubInterestAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.cornellappdev.coffee_chats_android
+
+import android.content.Context
+import android.graphics.PorterDuff
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.Filter
+import android.widget.Filterable
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.cornellappdev.coffee_chats_android.models.ClubOrInterest
+
+
+class ClubInterestAdapter(private val mContext: Context, list: Array<ClubOrInterest>, club: Boolean) :
+    ArrayAdapter<ClubOrInterest?>(mContext, 0, list), Filterable {
+    private var clubInterestList = list
+    private var isClubView = club
+    override fun getView(
+        position: Int,
+        convertView: View?,
+        parent: ViewGroup
+    ): View {
+        var listItem = convertView
+        if (isClubView) {
+            if (listItem == null) listItem =
+                LayoutInflater.from(mContext).inflate(R.layout.club_view, parent, false)
+        } else {
+            if (listItem == null) listItem =
+                LayoutInflater.from(mContext).inflate(R.layout.interest_view, parent, false)
+        }
+        val currentClubInterest = clubInterestList[position]
+        val clubOrInterestText = listItem!!.findViewById<TextView>(R.id.club_or_interest_text)
+        clubOrInterestText.setText(currentClubInterest.getText())
+        val clubOrInterestSubtext = listItem!!.findViewById<TextView>(R.id.club_or_interest_subtext)
+        clubOrInterestSubtext.setText(currentClubInterest.getSubtext())
```

Comment:
nit: here and above you can just do `clubOrInterestSubtext.text = ...`--no need for explicit setting/getting in kotlin